### PR TITLE
 Add flag to capitalize to lowercase remaining characters

### PR DESCRIPTION
**Pull request by ricardograca**
This adds a new flag to capitalize that when set to `true` will convert all characters after the first one to lowercase. Example:

    capitalize("FOO Bar", true);
    // => "Foo bar"

### CODE REVIEW

File: test/strings.js [hunk 1]
```diff
@@ -129,6 +129,7 @@ $(document).ready(function() {
     equal(_('fabio').capitalize(), 'Fabio', 'First letter is upper case');
     equal(_.capitalize('fabio'), 'Fabio', 'First letter is upper case');
     equal(_.capitalize('FOO'), 'FOO', 'Other letters unchanged');
+    equal(_.capitalize('FOO', true), 'Foo', 'Other letters are lowercased');
```
Comment (stoeffel) [inline test/strings.js]: could you add some more test i.e
```js
equal(_.capitalize('foO', true), 'Foo', 'Other letters are lowercased');

// and maybe some tests with lowercaseRest explicit "false"
```

Comment (ricardograca) [inline test/strings.js]: Sure.

Comment (stoeffel) [inline test/strings.js]: Cool, thanks.

**Comment by stoeffel** [comment]
Could you squash your commits to one? Would be nice, thanks.
[squashing commits](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html)

**Comment by ricardograca** [comment]
All of them? Because the Contributing document mentions providing separate commits with tests.

**Comment by stoeffel** [comment]
Yes, I would make one commit it's not really a bug. One commit would be cool.
Thanks and sorry for the nitpicking :smile:

**Comment by stoeffel** [comment]
:+1:

**Comment by ricardograca** [comment]
:joy:

**Pull request merged**